eth-contracts/test/serviceProvider.test.js

```diff
@@ -1,118 +1,95 @@
 
-    proxy = await OwnedUpgradeabilityProxy.new({ from: proxyOwner })
-
-    // Deploy registry
-    await registry.addContract(ownedUpgradeabilityProxyKey, proxy.address)
+  const [treasuryAddress, proxyAdminAddress, proxyDeployerAddress] = accounts
+  const proxyOwner = treasuryAddress
 
+  beforeEach(async () => {
     token = await AudiusToken.new({ from: treasuryAddress })
-    tokenAddress = token.address
-
-    let initialTokenBal = fromBn(await token.balanceOf(accounts[0]))
+    registry = await Registry.new({ from: treasuryAddress })
 
-    impl0 = await Staking.new()
-
-    // Create initialization data
-    let initializeData = encodeCall(
+    // Set up staking
+    staking0 = await Staking.new({ from: proxyAdminAddress })
+    stakingInitializeData = encodeCall(
       'initialize',
       ['address', 'address'],
-      [token.address, treasuryAddress])
-
-    // Initialize staking contract
-    await proxy.upgradeToAndCall(
-      impl0.address,
-      initializeData,
-      { from: proxyOwner })
-
+      [token.address, treasuryAddress]
+    )
+    proxy = await AdminUpgradeabilityProxy.new(
+      staking0.address,
+      proxyAdminAddress,
+      stakingInitializeData,
+      { from: proxyDeployerAddress }
+    )
     staking = await Staking.at(proxy.address)
-    stakingAddress = staking.address
+    await registry.addContract(stakingProxyKey, proxy.address, { from: treasuryAddress })
 
-    // Deploy sp storage
-    serviceProviderStorage = await ServiceProviderStorage.new(registry.address)
-    await registry.addContract(serviceProviderStorageKey, serviceProviderStorage.address)
+    // Deploy ServiceProviderStorage
+    serviceProviderStorage = await ServiceProviderStorage.new(registry.address, { from: treasuryAddress })
+    await registry.addContract(serviceProviderStorageKey, serviceProviderStorage.address, { from: treasuryAddress })
 
-    // Deploy sp factory
+    // Deploy ServiceProviderFactory
     serviceProviderFactory = await ServiceProviderFactory.new(
       registry.address,
-      ownedUpgradeabilityProxyKey,
-      serviceProviderStorageKey)
+      stakingProxyKey,
+      serviceProviderStorageKey,
+      { from: treasuryAddress }
+    )
 
-    await registry.addContract(serviceProviderFactoryKey, serviceProviderFactory.address)
+    await registry.addContract(serviceProviderFactoryKey, serviceProviderFactory.address, { from: treasuryAddress })
 
     // Permission sp factory as caller, from the proxy owner address
     // (which happens to equal treasury in this test case)
     await staking.setStakingOwnerAddress(serviceProviderFactory.address, { from: proxyOwner })
 
-    // Transfer 1000 tokens to accounts[1]
-    await token.transfer(accounts[1], INITIAL_BAL, { from: treasuryAddress })
+    // Transfer 1000 tokens to accounts[11]
+    await token.transfer(accounts[11], INITIAL_BAL, { from: treasuryAddress })
```

Comment:
same here, let's avoid using accounts >10 (index 9) in regular tests to fix CI

Reply:
fixed by changing CI config